scandium-core/src/test/java/org/eclipse/californium/scandium/DTLSConnectorAdvancedTest.java

```diff
@@ -1239,21 +1257,23 @@ public void setDrop(int drop) {
 		}
 
 		@Override
-		public void sendFlight(DTLSFlight flight) throws IOException {
+		public void sendFlight(DTLSFlight flight, Connection connection) throws IOException {
 			lastFlight = flight;
+			lastConnection = connection;
 			for (Record record : getMessagesOfFlight(flight)) {
 				connector.sendRecord(record.getPeerAddress(), record.toByteArray());
 			}
 		}
 
 		public void resendLastFlight() throws GeneralSecurityException, IOException {
 			DTLSFlight flight = lastFlight;
+			Connection connection = lastConnection;
 			if (null == flight) {
 				throw new IllegalStateException("no last flight!");
 			}
 			flight.incrementTries();
 			flight.setNewSequenceNumbers();
-			sendFlight(flight);
+			sendFlight(flight, connection);
```

Comment:
could be just : `sendFlight(flight, lastConnection)`

The idea is to ensure we alway send flight through the same connection  ?

Reply:
Yes. In one of the next versions the connection is get by address or cid, so I want to reduce the connection look ups.